tests/gdn/test_decode_vcache.py

```
@@ -0,0 +1,95 @@
+"""Correctness: gated_delta_rule_mtp_vcache_flush vs fp32 per-request recurrence.
+Draft outputs (at window rows [P:P+T]) and folded state (flushers -> fold of the
+P committed ring rows; verifiers -> unchanged) vs decode_delta_rule, per request."""
+import importlib.util, math, os, sys, torch
+REPO=os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, os.path.join(REPO,"tests","gdn"))
+from reference_delta_rule import decode_delta_rule
+def _load(n,p):
+    s=importlib.util.spec_from_file_location(n,p);m=importlib.util.module_from_spec(s);s.loader.exec_module(m);return m
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)